src/routes/csrf_protection.ts

```diff
@@ -1,13 +1,14 @@
 import { doubleCsrf } from "csrf-csrf";
 import sessionSecret from "../services/session_secret.js";
+import { isElectron } from "../services/utils.js";
 
 const doubleCsrfUtilities = doubleCsrf({
     getSecret: () => sessionSecret,
     cookieOptions: {
         path: "", // empty, so cookie is valid only for the current path
         secure: false,
         sameSite: "strict",
-        httpOnly: true
+        httpOnly: isElectron() ? false : true
```

Comment:
```suggestion
        httpOnly: !isElectron()
```

Reply:
yeah, would be a valid way as well, but:
It maybe sounds weird, but my feeling is that "isElectron() ? false : true" conveys the message a tiny bit better by being explicit?
I.e. cognitive load is a tiny bit lower this way.

e.g. if I read the code in my head, I actually read:

`isElectron() ? false : true` becomes => 
set httpOnly -> if it **is Electron** to **false** otherwise to **true**


`!isElectron()` becomes => 
set httpOnly -> to **is not Electron** 

which requires an extra second to think, IMHO :-)

Processing wise -> yes, !isElectron() would be "better", but this code is run exactly *once* at startup, so I'd say it's negligible :-)

not sure what your thoughts are? :-)

Reply:
@pano9000 , I suppose it's more of a matter of preference.

I've always "trimmed" redundant logic operations. Such as `b` instead of `b == true`, `!b`instead of `b == false`, and similarly with ternary operators.

In my opinion booleans are, by definition, with two possible values; so expressing directly whether it's `true` or not does not bring much value.

You can leave it as it is but I can't promise I won't sneak in at some point and change it. 😈

Reply:
I agree, let's agree on some middleground here then ;-)

What about:
```
// set httpOnly to false for Electron for now, see https://github.com/TriliumNext/Notes/pull/966
httpOnly: !isElectron()
```

Reply:
@pano9000 , LGTM.